packages/block-library/src/navigation/style.scss

```diff
@@ -122,27 +126,28 @@
 }
 
 // Default / Light styles
-.wp-block-navigation-link,
-.is-style-light .wp-block-navigation-link {
+.wp-block-navigation .wp-block-navigation__container .wp-block-navigation-link,
+.wp-block-navigation.is-style-light .wp-block-navigation__container .wp-block-navigation-link {
```

Comment:
`is-style-light` and `is-style-dark` need to apply to the whole navigation, not only to the submenus.